test/test_utils_hipgraphs.hpp

```diff
@@ -28,64 +28,53 @@
+
+                // Optionally launch the graph
+                if (launchGraph)
+                    HIP_CHECK_NON_VOID(hipGraphLaunch(graph_instance, stream));
+
+                // Optionally synchronize the stream when we're done
+                if (sync)
+                    HIP_CHECK_NON_VOID(hipStreamSynchronize(stream));
+            } 
     
-    inline hipGraph_t createGraphHelper(hipStream_t& stream, const bool beginCapture=true)
-    {
-        // Create a new graph
-        hipGraph_t graph;
-        HIP_CHECK_NON_VOID(hipGraphCreate(&graph, 0));
-
-        // Optionally begin stream capture
-        if (beginCapture)
-            HIP_CHECK_NON_VOID(hipStreamBeginCapture(stream, hipStreamCaptureModeGlobal));
-
-        return graph;
-    }
-
-    inline void cleanupGraphHelper(hipGraph_t& graph, hipGraphExec_t& instance)
-    {
-        HIP_CHECK_NON_VOID(hipGraphDestroy(graph));
-        HIP_CHECK_NON_VOID(hipGraphExecDestroy(instance));
-    }
-
-    inline void resetGraphHelper(hipGraph_t& graph, hipGraphExec_t& instance, hipStream_t& stream, const bool beginCapture=true)
-    {
-        // Destroy the old graph and instance
-        cleanupGraphHelper(graph, instance);
-
-        // Create a new graph and optionally begin capture
-        graph = createGraphHelper(stream, beginCapture);
-    }
-
-    inline hipGraphExec_t endCaptureGraphHelper(hipGraph_t& graph, hipStream_t& stream, const bool launchGraph=false, const bool sync=false)
-    {
-        // End the capture
-        HIP_CHECK_NON_VOID(hipStreamEndCapture(stream, &graph));
-
-        // Instantiate the graph
-        hipGraphExec_t instance;
-        HIP_CHECK_NON_VOID(hipGraphInstantiate(&instance, graph, nullptr, nullptr, 0));
-
-        // Optionally launch the graph
-        if (launchGraph)
-            HIP_CHECK_NON_VOID(hipGraphLaunch(instance, stream));
-
-        // Optionally synchronize the stream when we're done
-        if (sync)
-            HIP_CHECK_NON_VOID(hipStreamSynchronize(stream));
-
-        return instance;
-    }
-
-    inline void launchGraphHelper(hipGraphExec_t& instance, hipStream_t& stream, const bool sync=false)
-    {
-        HIP_CHECK_NON_VOID(hipGraphLaunch(instance, stream));
-
-        // Optionally sync after the launch
-        if (sync)
-            HIP_CHECK_NON_VOID(hipStreamSynchronize(stream));
-    }
-
+            inline void cleanupGraphHelper()
+            {
+                HIP_CHECK_NON_VOID(hipGraphDestroy(this->graph));
+                HIP_CHECK_NON_VOID(hipGraphExecDestroy(this->graph_instance));
+            }
+
+            inline void resetGraphHelper(hipStream_t& stream, const bool beginCapture=true)
+            {
+                // Destroy the old graph and instance
+                cleanupGraphHelper();
+
+                // Create a new graph and optionally begin capture
+                createGraph(stream, beginCapture);
```

Comment:
After we cleanup the old graph and instance (by calling cleanupGraphHelper above), we will need to restart the capture again. 

createGraph actually ends capture and launches the graph, so that's not really what we need here. If you make a new member function that calls hipStreamBeginCapture as per the comments above, I think you probably could call that here instead.

Reply:
Hy Wayne thanks for pointing my mistake out. I have added a startStreamCapture  in resetGraphHelper